routes/category.js

```diff
@@ -1,26 +1,21 @@
+
+
 const express = require('express');
 const router = express.Router();
-const products = require('../src/data/products');
-const categories = require('../src/data/categories');
+const service = require('../services/dataService');
 
 router.get('/:ctyId', function(req, res, next) {
-    const ctyId = req.params.ctyId;
-    let currentCategory;
-    let response = [];
-    categories.categories.forEach( category => {
-        if (category.id === ctyId) {
-            currentCategory = category;
-        }
-    });
-    products.products.forEach( product => {
-        if (product.categoryId === ctyId) {
-            response.push({
-                ...product,
-                category: currentCategory
-            });
-        }
-    });
-    res.send(response);
+  service.getProducts();
+  service.getCategories();
+  service.getCombinedProductMap();
+  const ctyId = req.params.ctyId;
+  const response = [];
+  for (const product in service.productMap) {
```

Comment:
Why you again call service.productMap? Assign the value to a variable in Line 10, and use it here. Refactor this method also

Reply:
I was trying to use the global variables which are readily available.